discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -451,102 +389,79 @@ impl SearchScope for SqliteStorage {
         documents: &[NewDocument],
     ) -> Result<(), Error> {
-        let mut tx = self.begin_tx().await?;
-        let mut query_builder = QueryBuilder::new("INSERT INTO ");
+        let mut tx = self.pool.begin().await?;
 
-        query_builder
-            .push(
-                "Search (rowid, searchBy, searchTerm, pageSize, pageNumber) VALUES (1, ?, ?, ?, ?)",
-            )
-            .build()
-            .persistent(false)
+        sqlx::query(
+            "INSERT INTO Search (rowid, searchBy, searchTerm, pageSize, pageNumber) VALUES (1, ?, ?, ?, ?)"
+        )
             .bind(search.search_by as u8)
             .bind(&search.search_term)
             .bind(search.paging.size)
             .bind(search.paging.next_page)
             .execute(&mut tx)
-            .await
-            .map_err(|err| Error::Database(err.into()))?;
+            .await?;
 
         if documents.is_empty() {
-            return Self::commit_tx(tx).await;
+            tx.commit().await?;
+            return Ok(());
         };
 
         SqliteStorage::store_new_documents(&mut tx, documents).await?;
-        query_builder
-            .reset()
-            .push("SearchDocument (documentId) ")
+
+        QueryBuilder::new("INSERT INTO SearchDocument (documentId) ")
             .push_values(documents, |mut stm, doc| {
                 stm.push_bind(&doc.id);
             })
             .build()
             .persistent(false)
             .execute(&mut tx)
-            .await
-            .map_err(|err| Error::Database(err.into()))?;
+            .await?;
 
-        Self::commit_tx(tx).await
+        tx.commit().await?;
+        Ok(())
     }
 
     async fn store_next_page(
         &self,
         page_number: u32,
         documents: &[NewDocument],
     ) -> Result<(), Error> {
-        let mut tx = self.begin_tx().await?;
-        let mut query_builder = QueryBuilder::new(String::new());
+        let mut tx = self.pool.begin().await?;
 
         SqliteStorage::store_new_documents(&mut tx, documents).await?;
-        query_builder
-            .push("INSERT INTO SearchDocument (documentId) ")
+
+        QueryBuilder::new("INSERT INTO SearchDocument (documentId) ")
             .push_values(documents, |mut stm, doc| {
                 stm.push_bind(&doc.id);
             })
             .build()
             .persistent(false)
             .execute(&mut tx)
-            .await
-            .map_err(|err| Error::Database(err.into()))?;
-        query_builder
-            .reset()
-            .push("UPDATE Search SET pageNumber = ? WHERE rowid = 1;")
-            .build()
-            .persistent(false)
+            .await?;
+
+        sqlx::query("UPDATE Search SET pageNumber = ? WHERE rowid = 1;")
             .bind(page_number)
             .execute(&mut tx)
-            .await
-            .map_err(|err| Error::Database(err.into()))?;
+            .await?;
 
-        Self::commit_tx(tx).await
+        tx.commit().await?;
+        Ok(())
```

Comment:
:art: this could be handled in a single expression
```suggestion
        tx.commit().await.map_err(Into::into)
```